rascal-textmate-core/src/main/rascal/lang/textmate/conversiontests/Walkthrough.rsc

```diff
@@ -0,0 +1,290 @@
+//   - extension 4: booleans.
+//
+// Working familiarity with TextMate grammar is assumed. To summarize:
+//
+//   - Syntax:
+//       - Each TextMate grammar consists of a list of TextMate rules (ordered).
+//       - Each TextMate rule is either a *match pattern* (consisting of one
+//         regular expression) or a *begin/end pattern* (consisting of two
+//         regular expressions and a list of nested TextMate rules).
+//
+//   - Semantics: A tokenization engine reads a document (line by line, top to
+//     bottom, left to right), while iteratively trying to apply TextMate rules
+//     by matching text against the regular expressions.
+//
+// Further reading:
+//   - https://macromates.com/manual/en/language_grammars
+//   - https://www.apeth.com/nonblog/stories/textmatebundle.html
+//   - https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide
+
+module lang::textmate::conversiontests::Walkthrough
+
+import Grammar;
+import ParseTree;
+import lang::textmate::Conversion;
+import lang::textmate::ConversionTests;
+
+import lang::rascal::grammar::analyze::Delimiters;
+
+// ## Basics
+//
+// The base fragment of the grammar looks as follows:
+
+lexical Alnum = [a-z A-Z 0-9];
+lexical Digit = [0-9];
+lexical Blank = [\ \t];
+lexical Space = [\ \t\n];
+
+start syntax Value
+    = Map
+    | Number
+    | String
+    | RegExp
+    | Location
+    | Boolean;
+
+syntax Map = "{" {(Key ":" Value) ","}* "}";
+
+lexical Key    = Alnum+ !>> [a-z A-Z 0-9];
+lexical Number = @category="constant.numeric" Digit+ !>> [0-9];
+lexical String = @category="string.quoted.double" "\"" Alnum* "\"";
+
+// Basically, the conversion algorithm converts each Rascal non-terminal that is
+// *suitable for conversion* to a TextMate rule. For instance, `Number` is
+// converted to the following match pattern (in JSON):
+//
+// ```
+// {
+//   "match": "([\\u0030-\\u0039]+?(?![\\u0030-\\u0039]))",
+//   "name": "prod(lex(\"Number\"),[conditional(iter(lex(\"Digit\")),{\\not-follow(\\char-class([range(48,57)]))})],{tag(\"category\"(\"constant.numeric\"))})", 
+//   "captures": {
+//     "1": {
+//       "name": "constant.numeric"
+//     }
+//   }
+// }
+// ```
+//
+// Note: The regular expression (`match` property) is written in Oniguruma
+// format (following the TextMate grammar specification), using code units
+// instead of alphanumericals.
+//
+// Note: The name (`name` property) could be anything, but to simplify
+// debugging, the conversion algorithm uses (part of) the internal
+// representation of the Rascal non-terminal as the name.
+//
+// In general, a Rascal non-terminal is "suitable for conversion" when it
+// satisfies each of the following conditions:
+//
+//  1. It is non-recursive. (Recursion is prohibitively hard to faithfully
+//     convert; tail-recursion could be workable, but currently out of scope.)
+//
+//  2. It does not match newlines. (TextMate rules that involve matching against
+//     newlines are dysfunctional.)
```

Comment:
nit: find different word for dysfunctional, either explain how they are hard to deal with, or do just say hard to support.